### DISCUSSION TITLE
How to properly use Kafka in the pytest?

### DISCUSSION THREAD

**Discussion by 1st** [confluentinc/confluent-kafka-python#1387] [Q&A]
How to properly use Kafka in the pytest?

We use the `confluent-kafka-python` in our codebase, and sometimes we have problems with message consumption (not all messages consumed, compared to what we produced by our code).

Do you have a manual or snippet of the best way to use Kafka in tests? Because I think about pathing the Kafka producer and consumer to be a python dicts (queue name -> list of messages), when it's problems with message deliveries, the happens randomly for our tests.

Will be happy if someone can explain the best practice how to use the patched version of the Kafka in the tests (we use pytest for tests).

**Comment by Tusenka** [comment]
have you found the solution?
